lib/enigma.rb

```diff
@@ -63,8 +71,24 @@ def decrypt_message(encrypted_message, key, date)
     decrypted
   end
 
+  def crack_message(encrypted_message, date)
+    cracked = ""
+    shifts = find_shifts(encrypted_message, date)
+    split_message(encrypted_message).each do |chars|
+      chars.zip(shifts).each do |char, shift_value|
+        if alphabet.include?(char)
+          new_index = (@alphabet.find_index(char) - shift_value) % 27
+          cracked.concat(@alphabet[new_index])
+        else
+          cracked.concat(char)
+        end
+      end
+    end
+    cracked
```

Comment:
Needs to be refactored. I never used the key to find the original message. just the shift values.